crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -57,11 +59,17 @@ impl Operations {
         None
     }
 
-    pub fn filter_by_topic(&self, topic_name: &str) -> Vec<(String, Operation)> {
+    pub fn filter_by_topic(
+        &self,
+        topic_name: &str,
+        prefix: &TopicPrefix,
+    ) -> Vec<(String, Operation)> {
         let mut vec: Vec<(String, Operation)> = Vec::new();
         for op in self.operations.iter() {
             match (op.topic(), op.on_fragment()) {
-                (None, Some(on_fragment)) => vec.push((on_fragment, op.clone())),
+                (None, Some(on_fragment)) if C8yDeviceControlTopic::name(prefix) == topic_name => {
+                    vec.push((on_fragment, op.clone()))
+                }
                 (Some(topic), Some(on_fragment)) if topic == topic_name => {
                     vec.push((on_fragment, op.clone()))
                 }
```

Comment:
An alternative way to fix the issue is to use `C8yDeviceControlTopic::name(prefix)` as a default value for `OnMessageExec::topic`. Doing so would be less error prone.

Reply:
In therory I agree but I cannot use it as serde default and changing `.topic()` function that unwraps to `C8yDeviceControlTopic::name(prefix)` when topic is `None` requires to provide that prefix almost in every single function in `operations.rs` (also in places that does not need that information) so I find it inconvenient.